src/log-store/src/kafka/log_store.rs

```diff
@@ -131,15 +139,54 @@ impl LogStore for KafkaLogStore {
             .raw_client
             .clone();
 
-        // Reads the entries starting from exactly the specified offset.
-        let offset = Offset::try_from(entry_id)?.0;
-        let mut stream_consumer = StreamConsumerBuilder::new(client, StartOffset::At(offset))
+        // Gets the offset of the lastest record in the topic. Actually, it's the lastest record of the single partition in the topic.
+        // The read operation terminates when this record is consumed.
+        let end_offset = client
+            .get_offset(OffsetAt::Latest)
+            .await
+            .context(GetOffsetSnafu { ns: ns.clone() })?;
+        // Reads entries with offsets in the range [start_offset, end_offset].
+        let start_offset = Offset::try_from(entry_id)?.0;
+
+        // Abort if there're no new entries.
+        // FIXME(niebayes): how come this case happens?
+        if start_offset > end_offset {
+            debug!("No new entries in ns {}", ns);
```

Comment:
It's better to change this log to info or warning and print the start_offset and end_offset to log too.